var/spack/repos/builtin/packages/pfunit/package.py

```diff
@@ -49,8 +49,8 @@ class Pfunit(CMakePackage):
     # See https://github.com/Goddard-Fortran-Ecosystem/pFUnit/pull/179
     conflicts("+shared", when="@4.0.0:")
     conflicts("+use_comm_world", when="~mpi")
-    conflicts('+mpi', when='@:3.99.99 %gcc@10.0.0:')
     patch("mpi-test.patch", when="@:3.99.99 +use_comm_world")
+    patch("pfunit-gfortran10-fix.patch", when="@:3.99.99 %gcc@10.0.0:")
```

Comment:
Couldn't we just add a `setup_build_environment` (see mpich) or `flag_handler` (see mvapich2) function, so we don't have to carry a patch?

Reply:
Unfortunately, pfunit's `CMakeLists.txt` overwrites `CMAKE_Fortran_FLAGS`, so it's not possible to set that from the outside. There may be hackier ways of doing this -- but this problem appears to be fixed in pfunit v4, and I wouldn't expect v3 to still see updates, so it doesn't seem likely that there's going to be much future change to this patch.

(I'm not really familiar, though, with the intricacies of how one can use the spack compiler wrappers to add additional flags. Though generally, I'm a believer in fixing brokenness where it occurs, not working around it from the outside. That's of course particularly true if the fix can be upstreamed, which I gotta admit I didn't even try here since I'm under the expression that pfunit v3 is superseded by v4)

Reply:
In general, the inject flags in `flag_handler` overwrite everything since they are set internally by the compiler wrappers. I agree that fixing the problem properly is preferable (especially if upstreamed) but that also means we would have to carry one more patch in the repository. I generally like to to keep the number of additional files in the repository low and use flags or URL patches if possible.

Reply:
Well, so I decided to go the way of (hopefully) minimal resistance and hacked the flag into the `package.py`. Ugly-ish, but it works.